daemon/cmd/daemon.go

```diff
@@ -611,6 +611,9 @@ func NewDaemon(ctx context.Context, epMgr *endpointmanager.EndpointManager, dp d
 		}
 	}
 
+	// Given BPF datapath is up, we can now remove old maps from version upgrade.
+	d.deleteMapsAfterUpgrade()
```

Comment:
On an upgrade from v1.7 to v1.8, we will:
1. Create `cilium_call_policy` from `initMaps()`.
2. Reload the base datapath (e.g., `bpf_overlay`, `bpf_network`, etc.). Some of these programs will start using the new and empty `cilium_call_policy` map.
3. Delete `cilium_policy`.
4. Trigger the regeneration/reload of endpoints. Once regenerated, endpoints will be added to `cilium_call_policy`.
5. Create and load the host endpoint (i.e., `bpf_host`). This program will start using the new `cilium_call_policy`.
6. All endpoints have been regenerated and `cilium_call_policy` is fully populated.

So, between steps 2 and 3, the base datapath programs will drop packets due to missed tail calls. Starting from step 3 and until all endpoints are regenerated, endpoints will be unreachable due to missed tail calls.

I think one way to fix it would be to rename `cilium_policy` to `cilium_call_policy` as a very first step before any regeneration.

Reply:
As discussed during sig-datapath with @pchaigno, rename sounds best option. @pchaigno will do a separate rename PR and I'll toss the rename part out of this fix.